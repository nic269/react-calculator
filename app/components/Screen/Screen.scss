@import 'app/styles/theme';

.container {
  background-color: $calculator-screen-bg-color;
  height: 95px;
}
