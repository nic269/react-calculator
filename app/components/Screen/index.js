/**
 *
 * Screen
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Textfit } from 'react-textfit';

import styles from './Screen.scss';

const Screen = ({ value }) => (
  <div className={styles.container}>
    <Textfit mode="single">{value}</Textfit>
  </div>
);

Screen.propTypes = {
  value: PropTypes.string,
};

Screen.defaultProps = {
  value: '',
};

export default memo(Screen);
