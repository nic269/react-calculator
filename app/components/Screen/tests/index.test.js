/**
 *
 * Tests for Screen
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';

import Screen from '../index';

describe('<Screen />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Screen />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Screen />);
    expect(firstChild).toMatchSnapshot();
  });
});
