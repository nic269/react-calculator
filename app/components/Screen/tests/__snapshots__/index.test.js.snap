// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> Should render and match the snapshot 1`] = `
<div>
  <div>
    <div
      style="display: inline-block; white-space: nowrap;"
    />
  </div>
</div>
`;
